src/views/Login.vue

```diff
@@ -1,39 +1,101 @@
 <template>
   <b-row>
-    <b-col>
-      <passthrough-form endpoint="/accounts/login/" :success-redirect-path="redirectPath" @formUpdated="onFormUpdated" />
-      <br />
-      <!-- TODO: Translate this -->
-      <p>Forgot your password?<router-link :to="{ name: 'passwordReset' }"> Reset it</router-link>.</p>
-      <!-- TODO: Translate this -->
-      <p>Not a member?<router-link :to="{ name: 'register' }"> Register</router-link>.</p>
-      <p>
-        Use of any LCO facilities and/or data implies acceptance of the
-        <a href="https://lco.global/observatory/termsofservice/"> Terms of Service</a>.
-      </p>
+    <b-col md="6" offset-md="3">
+      <b-card title="Login">
+        <b-form @submit.prevent="handleSubmit">
+          <b-form-group label="Username:" label-for="username">
+            <b-form-input id="username" v-model="form.username" type="text" required :disabled="loading" autofocus></b-form-input>
+          </b-form-group>
+
+          <b-form-group label="Password:" label-for="password">
+            <b-form-input id="password" v-model="form.password" type="password" required :disabled="loading"></b-form-input>
+          </b-form-group>
+
+          <b-alert v-if="error" variant="danger" show>
+            {{ error }}
+          </b-alert>
+
+          <b-button type="submit" variant="primary" :disabled="loading" class="w-100">
+            <b-spinner v-if="loading" small></b-spinner>
+            {{ loading ? 'Logging in...' : 'Login' }}
+          </b-button>
+        </b-form>
+      </b-card>
+      <div class="mt-3 text-center">
+        <!-- TODO: Translate this -->
+        <p>Forgot your password?<router-link :to="{ name: 'passwordReset' }"> Reset it</router-link>.</p>
+        <!-- TODO: Translate this -->
+        <p>Not a member?<router-link :to="{ name: 'register' }"> Register</router-link>.</p>
+        <p>
+          Use of any LCO facilities and/or data implies acceptance of the
+          <a href="https://lco.global/observatory/termsofservice/"> Terms of Service</a>.
+        </p>
+      </div>
     </b-col>
   </b-row>
 </template>
 <script>
-import PassthroughForm from '@/components/PassthroughForm.vue';
-
 export default {
   name: 'Login',
-  components: {
-    PassthroughForm
+  props: {
+    redirectPath: {
+      type: String,
+      default: null
+    }
   },
-  computed: {
-    redirectPath: function() {
-      return this.$route.query.next || '/';
+  data() {
+    return {
+      form: {
+        username: '',
+        password: ''
+      },
+      loading: false,
+      error: null
     }
   },
   methods: {
-    onFormUpdated: function(form) {
-      // Login form can also redirect to change password form.
-      // If the input button says "Change password" redirect to the
-      // correct Vue route. Otherwise this form will post to the wrong endpoint.
-      if (form.find("input[value='Change password']").length == 1)
```

Comment:
Is it okay to lose this behavior? I assume this is for if the password needs to be reset, then after login it would redirect them to the password reset form to change it?

Reply:
I've updated the front and backend to handle the password reset functionality.
I also discovered that in the current deployed version, if the password is expired it would still actually log the user in. That seems... bad? Let me know if I should preserve that.